### PR TITLE
Fix multilabel_accuracy of MixedHLabelAccuracy

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

https://jira.devtools.intel.com/browse/CVS-155228

https://github.com/Lightning-AI/torchmetrics/blob/6377aa5b6fe2863761839e6b8b5a857ef1b8acfa/src/torchmetrics/functional/classification/stat_scores.py#L583-L584

Due to the above code, an error is thrown if `num_multilabel_classes` is 1. To prevent this. (if num_multilabel_classes == 1, then use binary classification for that)

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/metrics/accuracy.py [hunk 1]
```diff
@@ -303,7 +305,7 @@ def _apply(self, fn: Callable, exclude_state: Sequence[str] = "") -> nn.Module:
             )
             for acc in self.multiclass_head_accuracy
         ]
-        if self.num_multilabel_classes > 0:
+        if hasattr(self, "multilabel_accuracy"):
```
Comment (eunwoosh) [inline src/otx/core/metrics/accuracy.py]: How about using `self.multilabel_accuracy == None` instead of it?

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work :) Could you add unit test for this case?

File: src/otx/core/metrics/accuracy.py (hunk 1 above)
Comment (harimkang) [reply src/otx/core/metrics/accuracy.py]: Is it okay to ask what the reason for this is? Is that a better way?

Comment (eunwoosh) [reply src/otx/core/metrics/accuracy.py]: Almost same but a little bit better thing to use it is that `self.multilabel_accuracy == None` is more explicit so it can deal with a case that attribute 'num_multilabel_classes' is set my mistake. If you want to leave it, you can ignore this comment :)

Comment (harimkang) [reply src/otx/core/metrics/accuracy.py]: i'll update it. you meant `if self.multilabel_accuracy is not None` right?

Comment (eunwoosh) [reply src/otx/core/metrics/accuracy.py]: Thanks. yes and also need to add `self.multilabel_accuracy = None` in the `__init__()`.

Comment (harimkang) [reply src/otx/core/metrics/accuracy.py]: Thanks! Done - https://github.com/openvinotoolkit/training_extensions/pull/4042/commits/f055880f5c8ce01d604e1922e93f8563c3a0f904

**Review by eunwoosh — APPROVED** [review]

**Review by chuneuny-emily — APPROVED** [review]
